juju/model.py

```diff
@@ -543,14 +567,20 @@ def _stop_watching(self):
         This coroutine blocks until the new object appears in the model.
 
         """
-        entity_added = asyncio.Queue(loop=self.loop)
+        return await self._wait(entity_type, entity_id, predicate)
```

Comment:
Shouldn't this be:

    return await self._wait(entity_type, entity_id, 'add', predicate)